### PR TITLE
[ENHANCEMENT] Sign in with branded button for external auth provider

### PR DESCRIPTION
**Pull request by celian-garcia**
# Description

# Screenshots

image
image

## Temporarily disabled the supported logos to see how it looks like when the logo is not yet supported.
image
image

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the
  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky.
  See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests)
  and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues
  include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

File: ui/app/src/views/auth/SignWrapper.tsx [hunk 1]
```diff
@@ -11,26 +11,51 @@
...
+    return MicrosoftLoginButton;
+  }
+  if (url.includes('github')) {
+    return GithubLoginButton;
+  }
+  if (url.includes('google')) {
+    return GoogleLoginButton;
+  }
+  //TODO: Support more providers managed by the lib
+  //TODO: Support more providers not managed by the lib (e.g Gitlab)
```
Comment (Nexucis) [inline ui/app/src/views/auth/SignWrapper.tsx:41]: we could try to propose a new button to the upstream lib. WDYT @celian-garcia ?

Comment (celian-garcia) [reply ui/app/src/views/auth/SignWrapper.tsx:41]: Yes indeed that'd probably be well approved.

Comment (celian-garcia) [reply ui/app/src/views/auth/SignWrapper.tsx]: PR proposed for Gitlab https://github.com/michaeljscript/react-social-login-buttons/pull/115

File: ui/app/src/views/auth/SignWrapper.tsx [hunk 2]
```diff
@@ -52,22 +150,26 @@ export function SignWrapper(props: { children: ReactNode }) {
       />
       <Stack gap={1} sx={{ maxWidth: '85%', minWidth: '200px' }}>
         {nativeProviderIsEnabled && props.children}
-        {socialProviders.length > 0 && (
+        {nativeProviderIsEnabled && socialProviders.length > 0 && (
           <div>
-            <Divider sx={{ marginTop: '16px' }}>{nativeProviderIsEnabled && 'or'} sign in with</Divider>
+            <Divider sx={{ marginTop: '16px' }}>{nativeProviderIsEnabled && 'or'}</Divider>
```
Comment (Gladorme) [inline ui/app/src/views/auth/SignWrapper.tsx:155]:
```suggestion
            <Divider sx={{ marginTop: '16px' }}>or</Divider>
```

Comment (celian-garcia) [reply ui/app/src/views/auth/SignWrapper.tsx:155]: The idea is to have the  'or' only if the nativeProvider is enabled. Can we talk about what you mean ?

Comment (celian-garcia) [reply ui/app/src/views/auth/SignWrapper.tsx:155]: aaah okay I understood, let me try

**Review by Gladorme — APPROVED** [review]

**Pull request merged**